packages/components/inputs/localized-rich-text-input/src/editor.tsx

```diff
@@ -161,34 +208,76 @@ const Editor = (props: TEditorProps) => {
-                hasError={props.hasError}
-                isDisabled={props.isDisabled}
-                hasWarning={props.hasWarning}
-                isReadOnly={Boolean(props.isReadOnly)}
-                editor={props.editor as TEditor}
-                containerStyles={containerStyles}
-                showExpandIcon={props.showExpandIcon}
-                onClickExpand={props.onClickExpand}
+              <Slate
+                editor={editor}
+                value={props.value}
+                onChange={props.onChange}
               >
-                {props.children}
-              </RichTextBody>
+                <EditorLanguageLabel htmlFor={props.id} theme={theme}>
+                  {/* FIXME: add proper tone for disabled when tones are refactored */}
+                  <Text.Detail tone="secondary">
+                    {props.language.toUpperCase()}
+                  </Text.Detail>
+                </EditorLanguageLabel>
+                <RichTextBody
+                  // @ts-ignore
+                  ref={refObj as unknown as Ref<TRichTextEditorBodyRef>}
+                  styles={{
+                    container: css`
+                      flex: auto;
+                      width: 0;
+                      border-top-left-radius: 0;
+                      border-bottom-left-radius: 0;
+                    `,
+                  }}
+                  hasError={props.hasError}
+                  isDisabled={props.isDisabled}
+                  hasWarning={props.hasWarning}
+                  isReadOnly={Boolean(props.isReadOnly)}
+                  showExpandIcon={Boolean(props.showExpandIcon)}
+                  onClickExpand={props.onClickExpand}
+                  containerStyles={containerStyles}
+                >
+                  <Editable
+                    {...filterDataAttributes(props)}
+                    name={props.name}
+                    renderElement={renderElement}
+                    renderLeaf={renderLeaf}
+                    placeholder={props.placeholder}
+                    autoFocus={props.isAutofocused}
+                    onBlur={props.onBlur}
+                    onFocus={(event) => {
+                      props.onFocus?.(event);
+                      // opens the input if it regains focus and it's closed
+                      if (!isOpen) {
+                        toggle();
+                        focusEditor(editor);
+                      }
+                    }}
+                    readOnly={props.isReadOnly}
+                    disabled={props.isDisabled}
+                    onKeyDown={(event) => {
+                      for (const hotkey in HOTKEYS) {
+                        if (isHotkey(hotkey, event)) {
+                          event.preventDefault();
+                          const mark = HOTKEYS[hotkey as keyof typeof HOTKEYS];
+                          toggleMark(editor, mark);
+                        }
+                      }
+                    }}
```

Comment:
nit: Although the HOTKEYS object currently only have 4 attributes, we are iterating over all of them for every keystroke.

Maybe we could break after finding a hot key:
```suggestion
                  onKeyDown={(event) => {
                    for (const hotkey in HOTKEYS) {
                      if (isHotkey(hotkey, event)) {
                        event.preventDefault();
                        const mark = HOTKEYS[hotkey as keyof typeof HOTKEYS];
                        toggleMark(editor, mark);
                        break;
                      }
                    }
                  }}
```

Reply:
nice improvement!